src/bin/workload_generator.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Anza Technology, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::net::{SocketAddr, UdpSocket};
+use std::time::{Duration, Instant};
+
+use alpenglow::crypto::aggsig;
+use alpenglow::crypto::signature::SecretKey;
+use alpenglow::network::NetworkMessage;
+use alpenglow::{Transaction, ValidatorInfo, logging};
+use clap::Parser;
+use color_eyre::Result;
+use log::info;
+use rand::RngCore;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct ConfigFile {
+    id: u64,
+    identity_key: SecretKey,
+    #[serde(deserialize_with = "aggsig::SecretKey::from_array_of_bytes")]
+    voting_key: aggsig::SecretKey,
+    port: u16,
+    gossip: Vec<ValidatorInfo>,
+}
+
+/// Worklaod generator for benchmarks.
+#[derive(Debug, Clone, Parser)]
+#[command(version, about, long_about = None)]
+struct Args {
+    /// Validator addres or receiving transactions.
+    #[arg(long)]
+    validator: String,
+    /// Initial wait time, before sending any transactions.
+    #[arg(long)]
+    initial_delay_secs: Option<u64>,
+    /// Target throughput in transactions per second.
+    #[arg(long)]
+    transactions_per_second: Option<u64>,
+    /// Bytes per transaction.
+    #[arg(long)]
+    transaction_size: Option<u64>,
+}
+
+#[tokio::main]
```

Comment:
I suppose this doesn't need to be async at the moment (we are just using sleep which we could do without async) but will probably in the near future when we want to produce transactions from many threads.